runtime/onert/core/src/exec/Execution.cc

```diff
@@ -32,6 +32,15 @@ Execution::Execution(const std::shared_ptr<ExecutorMap> &executors) : _executors
   _io_desc.outputs.resize(primary_subg.getOutputs().size());
 }
 
+void Execution::assertAsyncInactive()
+{
+  assert(!_exec_thread || !_exec_thread->joinable());
+  if (_exec_thread && _exec_thread->joinable())
```

Comment:
Here it checks the same condition right after an assertion. we usually don't do something like this, as assertions works only for debug mode and used to catch ONE developer's fault early, not the user's. Can I ask what your intention is here?

Reply:
Intention is to trigger assert in debug mode and silently fix the problem in release mode.